compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1199,6 +1199,7 @@ class Namer { typer: Typer =>
       cls.setNoInitsFlags(parentsKind(parents), untpd.bodyKind(rest))
       if (cls.isNoInitsClass) cls.primaryConstructor.setFlag(StableRealizable)
       processExports(using localCtx)
+      defn.patchStdLibClass(denot.asClass)
```

Comment:
What happens when we pickle Predef.scala to Tasty, then unpickle it? As far as I can tell, since we're only adding symbols and not trees, the pickled Predef won't have the extra definitions, so we need to run `patchStdlibClass` in TreeUnpickler too.

Reply:
But do we ever unpickle Predef? Why would we want to do that?

Reply:
The doctool should probably see the unpickled Predef, for definition lookup purposes. I'd also worry about any user of Tasty Inspector that wants to look up definitions in `Predef`.

Reply:
But so far nobody pickles Predef into Tasty, right?

Reply:
I checked and it seems a `Predef.tasty` file is created, we also document it in the doctool already:
https://scala3doc.virtuslab.com/pr-master/scala3-stdlib/api/scala/-predef/index.html?query=%20object%20Predef%20extends%20LowPriorityImplicits